pkg/storage/tsdb/tenant_deletion_mark.go

```diff
@@ -0,0 +1,39 @@
+package tsdb
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"path"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/thanos-io/thanos/pkg/objstore"
+)
+
+const TenantDeletionMarkFilename = "marker/tenant-deletion-mark.json"
```

Comment:
>  I don't remember if it's safe with regards of filesystem-based bucket.

If it's not, I'd argue that filesystem-bucket is broken and should be fixed. Since we want to use `markers` prefix, we need to use some separator, and we already use `/` elsewhere. (E.g. `UserBucketClient` uses it in its `fullName` method)

Also renamed "filename" to path.